### PR TITLE
Code release

### PR DESCRIPTION
**Pull request by hiyyg**

### CODE REVIEW

**Comment by hiyyg** [comment]
time?

**Comment by FengZicai** [comment]
Hi,

I apologize for the delayed response. I am currently working on a new submission, and the code requires some minor modifications and organization. I will release the code as soon as possible.

Thank you for your understanding.

**Comment by MAI-KOKO** [comment]
> Hi,
> 
> I apologize for the delayed response. I am currently working on a new submission, and the code requires some minor modifications and organization. I will release the code as soon as possible.对于延迟回复，我深表歉意。我目前正在处理一个新的提交，代码需要一些小的修改和组织。我会尽快发布代码。
> 
> Thank you for your understanding.

Hi, your work is excellent! May I ask when you expect to release the code?

**Comment by laikakano** [comment]
Hi, When I was validating on the nuscenes dataset, I encountered two issues and I hope you can help me check them. The first issue is that I couldn't find the path mentioned in the README during the environment setup, and when I directly ran CMake in the src directory, it resulted in an error. The second issue is that the inference results on the nuscenes validation set are as follows. Could you please advise on how to resolve these issues?

Total 168532 scenes in the val split
********************************************************************************
Test network performance on validation split
********************************************************************************
100%|█████████████████████████████████████| 6019/6019 [5:45:00<00:00,  6.66s/it]Validation per class iou: 
barrier : 0.00%
bicycle : 0.00%
bus : 0.00%
car : 0.00%
construction_vehicle : 0.00%
motorcycle : 0.00%
pedestrian : 0.00%
traffic_cone : 0.00%
trailer : 0.00%
truck : 0.00%
driveable_surface : 0.00%
other_flat : 0.00%
sidewalk : 0.01%
terrain : 0.20%
manmade : 16.36%
vegetation : 28.65%
100%|█████████████████████████████████████| 6019/6019 [5:45:00<00:00,  3.44s/it]
Current val miou is 2.827 
Inference time per 28 is 0.4535 seconds

**Comment by xiaosa269** [comment]
> Hi, When I was validating on the nuscenes dataset, I encountered two issues and I hope you can help me check them. The first issue is that I couldn't find the path mentioned in the README during the environment setup, and when I directly ran CMake in the src directory, it resulted in an error. The second issue is that the inference results on the nuscenes validation set are as follows. Could you please advise on how to resolve these issues?
> 
> Total 168532 scenes in the val split
> 
> Test network performance on validation split
> 
> 100%|█████████████████████████████████████| 6019/6019 [5:45:00<00:00, 6.66s/it]Validation per class iou: barrier : 0.00% bicycle : 0.00% bus : 0.00% car : 0.00% construction_vehicle : 0.00% motorcycle : 0.00% pedestrian : 0.00% traffic_cone : 0.00% trailer : 0.00% truck : 0.00% driveable_surface : 0.00% other_flat : 0.00% sidewalk : 0.01% terrain : 0.20% manmade : 16.36% vegetation : 28.65% 100%|█████████████████████████████████████| 6019/6019 [5:45:00<00:00, 3.44s/it] Current val miou is 2.827 Inference time per 28 is 0.4535 seconds

**Comment by xiaosa269** [comment]
> Hi, When I was validating on the nuscenes dataset, I encountered two issues and I hope you can help me check them. The first issue is that I couldn't find the path mentioned in the README during the environment setup, and when I directly ran CMake in the src directory, it resulted in an error. The second issue is that the inference results on the nuscenes validation set are as follows. Could you please advise on how to resolve these issues?
> 
> Total 168532 scenes in the val split
> 
> Test network performance on validation split
> 
> 100%|█████████████████████████████████████| 6019/6019 [5:45:00<00:00, 6.66s/it]Validation per class iou: barrier : 0.00% bicycle : 0.00% bus : 0.00% car : 0.00% construction_vehicle : 0.00% motorcycle : 0.00% pedestrian : 0.00% traffic_cone : 0.00% trailer : 0.00% truck : 0.00% driveable_surface : 0.00% other_flat : 0.00% sidewalk : 0.01% terrain : 0.20% manmade : 16.36% vegetation : 28.65% 100%|█████████████████████████████████████| 6019/6019 [5:45:00<00:00, 3.44s/it] Current val miou is 2.827 Inference time per 28 is 0.4535 seconds

Hello, have you managed to resolve this issue? I’m encountering the same problem when testing the SemanticKITTI validation set with the provided weights, and I’ve noticed that some model parameters don’t seem to load correctly. Have you run into a similar issue?


test_skitti.py
{'format_version': 1, 'model_params': {'model_architecture': 'largekernelseg', 'input_dims': 13, 'spatial_shape': [2000, 2000, 120], 'scale_list': [2, 4, 8, 16], 'hiden_size': 64, 'num_classes': 20, 'large_kernel_size': [9, 9, 9], 'spatial_group_partition': [0, 3, 6, 9], 'model_load_path': 'output_skitti/opensource_9ks_s030_w64_0.pt', 'model_save_path': 'output_skitti/opensource_9ks_s030_w64_.pt'}, 'dataset_params': {'training_size': 19130, 'dataset_type': 'point_semkitti_mix', 'pc_dataset_type': 'SemKITTI_sk', 'collate_type': 'mix_collate_fn_default', 'ignore_label': 0, 'label_mapping': './config/label_mapping/semantic-kitti-sub.yaml', 'num_classes': 20, 'spatial_shape': [2000, 2000, 120], 'max_volume_space': [50, 50, 2], 'min_volume_space': [-50, -50, -4], 'seg_labelweights': [0, 55437630, 320797, 541736, 2578735, 3274484, 552662, 184064, 78858, 240942562, 17294618, 170599734, 6369672, 230413074, 101130274, 476491114, 9833174, 129609852, 4506626, 1168181], 'train_data_loader': {'data_path': '/data/xhy/semantickitti/dataset/sequences/', 'imageset': 'train', 'batch_size': 2, 'num_workers': 4, 'rotate_aug': True, 'flip_aug': True, 'scale_aug': True, 'transform_aug': True, 'dropout_aug': True, 'polarmix_aug': False, 'mix_aug': True, 'instance_aug': False, 'd_point_num': 80000}, 'val_data_loader': {'data_path': '/data/xhy/semantickitti/dataset/sequences/', 'num_workers': 14, 'imageset': 'val', 'batch_size': 14, 'rotate_aug': True, 'flip_aug': True, 'scale_aug': True, 'transform_aug': True, 'dropout_aug': False, 'polarmix_aug': False, 'mix_aug': False, 'd_point_num': -1}}, 'train_params': {'seed': 1588147245, 'max_num_epochs': 30, 'learning_rate': 0.0002, 'optimizer': 'SGD', 'lr_scheduler': 'StepLR', 'momentum': 0.9, 'weight_decay': 0.0001, 'nesterov': True, 'lambda_lovasz': 1, 'decay_step': 1, 'decay_rate': 0.5, 'eval_every_n_steps': 299, 'distributed': False, 'amp_enabled': False}, 'sparse_params': {'use_sparse': True, 'growth': 'random', 'prune': 'magnitude', 'redistribution': 'none', 'prune_rate': 0.3, 'sparsity': 0.3, 'sparse_init': 'ERK', 'update_frequency': 8000, 'stop_sparse_epoch': 0}, 'config_path': './config/lk-semantickitti_erk_finetune.yaml', 'ip': '127.0.0.1', 'port': '3023', 'num_vote': 14}
pre-train
not matched key spv_enc.0.v_enc.0.layers.0.weight
not matched key spv_enc.0.v_enc.0.layers.3.weight
not matched key spv_enc.0.v_enc.1.layers.0.weight
not matched key spv_enc.0.v_enc.1.layers.3.weight
not matched key spv_enc.1.v_enc.0.layers.0.weight
not matched key spv_enc.1.v_enc.0.layers.3.weight
not matched key spv_enc.1.v_enc.1.layers.0.weight
not matched key spv_enc.1.v_enc.1.layers.3.weight
not matched key spv_enc.2.v_enc.0.layers.0.weight
not matched key spv_enc.2.v_enc.0.layers.3.weight
not matched key spv_enc.2.v_enc.1.layers.0.weight
not matched key spv_enc.2.v_enc.1.layers.3.weight
not matched key spv_enc.3.v_enc.0.layers.0.weight
not matched key spv_enc.3.v_enc.0.layers.3.weight
not matched key spv_enc.3.v_enc.1.layers.0.weight
not matched key spv_enc.3.v_enc.1.layers.3.weight
matched parameter sets: 241, and no matched: 16

100%|█████████████████████████████████████| 4071/4071 [5:45:58<00:00,  1.59s/it]Validation per class iou: 
car : 0.54%
bicycle : 0.00%
motorcycle : 0.00%
truck : 0.00%
bus : 0.00%
person : 0.00%
bicyclist : 0.00%
motorcyclist : 0.00%
road : 0.00%
parking : 0.00%
sidewalk : 3.11%
other-ground : 0.00%
building : 0.76%
fence : 0.00%
vegetation : 31.83%
trunk : 0.00%
terrain : 0.25%
pole : 0.00%
traffic-sign : 18.86%
100%|█████████████████████████████████████| 4071/4071 [5:45:58<00:00,  5.10s/it]
Current val miou is 2.913 
Inference time per 14 is 1.5664 seconds

**Comment by faithzxr** [comment]
> Hi, When I was validating on the nuscenes dataset, I encountered two issues and I hope you can help me check them. The first issue is that I couldn't find the path mentioned in the README during the environment setup, and when I directly ran CMake in the src directory, it resulted in an error. The second issue is that the inference results on the nuscenes validation set are as follows. Could you please advise on how to resolve these issues?
> 
> Total 168532 scenes in the val split
> 
> Test network performance on validation split
> 
> 100%|█████████████████████████████████████| 6019/6019 [5:45:00<00:00, 6.66s/it]Validation per class iou: barrier : 0.00% bicycle : 0.00% bus : 0.00% car : 0.00% construction_vehicle : 0.00% motorcycle : 0.00% pedestrian : 0.00% traffic_cone : 0.00% trailer : 0.00% truck : 0.00% driveable_surface : 0.00% other_flat : 0.00% sidewalk : 0.01% terrain : 0.20% manmade : 16.36% vegetation : 28.65% 100%|█████████████████████████████████████| 6019/6019 [5:45:00<00:00, 3.44s/it] Current val miou is 2.827 Inference time per 28 is 0.4535 seconds

May I ask if you have solved it? I have the same problem

**Comment by faithzxr** [comment]
> Hi, When I was validating on the nuscenes dataset, I encountered two issues and I hope you can help me check them. The first issue is that I couldn't find the path mentioned in the README during the environment setup, and when I directly ran CMake in the src directory, it resulted in an error. The second issue is that the inference results on the nuscenes validation set are as follows. Could you please advise on how to resolve these issues?
May I ask if you have solved it? I also encountered your second problem，Thanks

**Comment by lemon1022** [comment]
> > Hi, When I was validating on the nuscenes dataset, I encountered two issues and I hope you can help me check them. The first issue is that I couldn't find the path mentioned in the README during the environment setup, and when I directly ran CMake in the src directory, it resulted in an error. The second issue is that the inference results on the nuscenes validation set are as follows. Could you please advise on how to resolve these issues?
> > Total 168532 scenes in the val split
> > Test network performance on validation split
> > 100%|█████████████████████████████████████| 6019/6019 [5:45:00<00:00, 6.66s/it]Validation per class iou: barrier : 0.00% bicycle : 0.00% bus : 0.00% car : 0.00% construction_vehicle : 0.00% motorcycle : 0.00% pedestrian : 0.00% traffic_cone : 0.00% trailer : 0.00% truck : 0.00% driveable_surface : 0.00% other_flat : 0.00% sidewalk : 0.01% terrain : 0.20% manmade : 16.36% vegetation : 28.65% 100%|█████████████████████████████████████| 6019/6019 [5:45:00<00:00, 3.44s/it] Current val miou is 2.827 Inference time per 28 is 0.4535 seconds
> 
> Hello, have you managed to resolve this issue? I’m encountering the same problem when testing the SemanticKITTI validation set with the provided weights, and I’ve noticed that some model parameters don’t seem to load correctly. Have you run into a similar issue?
> 
> test_skitti.py {'format_version': 1, 'model_params': {'model_architecture': 'largekernelseg', 'input_dims': 13, 'spatial_shape': [2000, 2000, 120], 'scale_list': [2, 4, 8, 16], 'hiden_size': 64, 'num_classes': 20, 'large_kernel_size': [9, 9, 9], 'spatial_group_partition': [0, 3, 6, 9], 'model_load_path': 'output_skitti/opensource_9ks_s030_w64_0.pt', 'model_save_path': 'output_skitti/opensource_9ks_s030_w64_.pt'}, 'dataset_params': {'training_size': 19130, 'dataset_type': 'point_semkitti_mix', 'pc_dataset_type': 'SemKITTI_sk', 'collate_type': 'mix_collate_fn_default', 'ignore_label': 0, 'label_mapping': './config/label_mapping/semantic-kitti-sub.yaml', 'num_classes': 20, 'spatial_shape': [2000, 2000, 120], 'max_volume_space': [50, 50, 2], 'min_volume_space': [-50, -50, -4], 'seg_labelweights': [0, 55437630, 320797, 541736, 2578735, 3274484, 552662, 184064, 78858, 240942562, 17294618, 170599734, 6369672, 230413074, 101130274, 476491114, 9833174, 129609852, 4506626, 1168181], 'train_data_loader': {'data_path': '/data/xhy/semantickitti/dataset/sequences/', 'imageset': 'train', 'batch_size': 2, 'num_workers': 4, 'rotate_aug': True, 'flip_aug': True, 'scale_aug': True, 'transform_aug': True, 'dropout_aug': True, 'polarmix_aug': False, 'mix_aug': True, 'instance_aug': False, 'd_point_num': 80000}, 'val_data_loader': {'data_path': '/data/xhy/semantickitti/dataset/sequences/', 'num_workers': 14, 'imageset': 'val', 'batch_size': 14, 'rotate_aug': True, 'flip_aug': True, 'scale_aug': True, 'transform_aug': True, 'dropout_aug': False, 'polarmix_aug': False, 'mix_aug': False, 'd_point_num': -1}}, 'train_params': {'seed': 1588147245, 'max_num_epochs': 30, 'learning_rate': 0.0002, 'optimizer': 'SGD', 'lr_scheduler': 'StepLR', 'momentum': 0.9, 'weight_decay': 0.0001, 'nesterov': True, 'lambda_lovasz': 1, 'decay_step': 1, 'decay_rate': 0.5, 'eval_every_n_steps': 299, 'distributed': False, 'amp_enabled': False}, 'sparse_params': {'use_sparse': True, 'growth': 'random', 'prune': 'magnitude', 'redistribution': 'none', 'prune_rate': 0.3, 'sparsity': 0.3, 'sparse_init': 'ERK', 'update_frequency': 8000, 'stop_sparse_epoch': 0}, 'config_path': './config/lk-semantickitti_erk_finetune.yaml', 'ip': '127.0.0.1', 'port': '3023', 'num_vote': 14} pre-train not matched key spv_enc.0.v_enc.0.layers.0.weight not matched key spv_enc.0.v_enc.0.layers.3.weight not matched key spv_enc.0.v_enc.1.layers.0.weight not matched key spv_enc.0.v_enc.1.layers.3.weight not matched key spv_enc.1.v_enc.0.layers.0.weight not matched key spv_enc.1.v_enc.0.layers.3.weight not matched key spv_enc.1.v_enc.1.layers.0.weight not matched key spv_enc.1.v_enc.1.layers.3.weight not matched key spv_enc.2.v_enc.0.layers.0.weight not matched key spv_enc.2.v_enc.0.layers.3.weight not matched key spv_enc.2.v_enc.1.layers.0.weight not matched key spv_enc.2.v_enc.1.layers.3.weight not matched key spv_enc.3.v_enc.0.layers.0.weight not matched key spv_enc.3.v_enc.0.layers.3.weight not matched key spv_enc.3.v_enc.1.layers.0.weight not matched key spv_enc.3.v_enc.1.layers.3.weight matched parameter sets: 241, and no matched: 16
> 
> 100%|█████████████████████████████████████| 4071/4071 [5:45:58<00:00, 1.59s/it]Validation per class iou: car : 0.54% bicycle : 0.00% motorcycle : 0.00% truck : 0.00% bus : 0.00% person : 0.00% bicyclist : 0.00% motorcyclist : 0.00% road : 0.00% parking : 0.00% sidewalk : 3.11% other-ground : 0.00% building : 0.76% fence : 0.00% vegetation : 31.83% trunk : 0.00% terrain : 0.25% pole : 0.00% traffic-sign : 18.86% 100%|█████████████████████████████████████| 4071/4071 [5:45:58<00:00, 5.10s/it] Current val miou is 2.913 Inference time per 14 is 1.5664 seconds

Hi! I have the same problem! May I ask if you have solved it?

**Comment by amokame** [comment]
A small question. Why the input_dims set to 13? I thought it is the input feature’s channel
